test/test_namedtensor.py

```diff
@@ -183,6 +183,14 @@ def test_no_multiprocessing_support(self):
         with self.assertRaisesRegex(RuntimeError, "NYI"):
             ForkingPickler(buf, pickle.HIGHEST_PROTOCOL).dump(named_tensor)
 
+    def test_big_tensor_repr(self):
+        def check_repr(named_tensor):
+            unnamed_tensor = named_tensor.view_names(None)
+            expected = "{}, names={})".format(repr(unnamed_tensor)[:-1], named_tensor.names)
+            self.assertEqual(repr(named_tensor), expected)
+
+        check_repr(torch.randn(128, 3, 64, 64, names=('N', 'C', 'H', 'W')))
```

Comment:
I think the test in this case would have passed with or without your change to `_tensor_str`, what's the case where the names will now be missing from the repr?

Reply:
The test case crashes with `RuntimeError: cat: no named inference rule implemented.`. We use a lot of various ops to format the tensor's data into a form that can be printed. If the tensor's data is too large to fit on screen, it needs to be summarized, which is where the `torch.cat` call came from.

Reply:
ah you mean to generate the truncated version of the tensor data? maybe we can make that comment below clearer, the meaning wasn't obvious to me.

Reply:
Done